resources/schema.py

```diff
@@ -446,41 +447,12 @@ class BerthInput(AbstractBoatPlaceInput):
     pier_id = graphene.ID()
     comment = graphene.String()
     is_accessible = graphene.Boolean()
-    berth_type_id = graphene.ID()
     width = graphene.Float(description=_("width (m)"))
     length = graphene.Float(description=_("length (m)"))
     depth = graphene.Float(description=_("depth (m)"))
     mooring_type = BerthMooringTypeEnum()
 
 
-def get_berth_type(info, input):
-    berth_type = None
-    berth_type_id = input.pop("berth_type_id", None)
-    width = input.pop("width", None)
-    length = input.pop("length", None)
-    depth = input.pop("depth", None)
-    mooring_type = input.pop("mooring_type", None)
-    berth_type_params = [width, length, depth, mooring_type]
-
-    # Cannot have both fields at the same time
-    if berth_type_id and any(berth_type_params):
-        raise VenepaikkaGraphQLError(_("Cannot receive BerthType and dimensions"))
-
-    if berth_type_id:
-        berth_type = get_node_from_global_id(
-            info, berth_type_id, only_type=BerthTypeNode, nullable=False,
-        )
-    elif any(berth_type_params):
-        if all(berth_type_params):
-            berth_type, created = BerthType.objects.get_or_create(
-                width=width, length=length, depth=depth, mooring_type=mooring_type
-            )
-        else:
-            raise VenepaikkaGraphQLError(_("Missing fields to associate a BerthType"))
-
-    return berth_type
-
-
 class CreateBerthMutation(graphene.ClientIDMutation):
     class Input(BerthInput):
         number = graphene.Int(required=True)
```

Comment:
`CreateBerthMutationInput` should have `width`, `length` and `mooringType` as `required=True`

Reply:
Good catch! This also fixes the other comment.

<img width="344" alt="image" src="https://user-images.githubusercontent.com/15201480/80592642-72594780-8a28-11ea-9230-5ab0aed6e9f2.png">